src/main/java/cz/muni/fi/pv286/mmt/arguments/ArgParser.java

```diff
@@ -0,0 +1,504 @@
+        }
+        return options[index + 1];
+    }
+
+    private static String extractOption(String option)
+            throws IllegalArgumentException, IndexOutOfBoundsException {
+        Matcher m = Patterns.extractor.matcher(option);
+        if (m.find()) {
+            return m.group(1);
+        } else {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    private static void checkHasInputFormat(Options options)
+            throws BadArgumentsException {
+        if (options.getInputFormat() == null || options.getOutputFormat() == null) {
+            throw new BadArgumentsException(MESSAGE_INVALID_ARGUMENTS);
+        }
+    }
+
+    private static void checkHasInputFromToOptionAndIsRightFormat(Options options)
+            throws BadArgumentsException {
+        if (options.getInputFromToOption().isPresent()
+                && (options.getInputFormat() != IoFormat.INT
+                && options.getInputFormat() != IoFormat.BITS)) {
+            throw new BadArgumentsException(MESSAGE_INVALID_ARGUMENTS);
+        }
+    }
+
+    private static void checkHasOutputFromToOptionAndIsRightFormat(Options options)
+            throws BadArgumentsException {
+        if (options.getOutputFromToOption().isPresent()
+                && (options.getOutputFormat() != IoFormat.INT
+                && options.getOutputFormat() != IoFormat.ARRAY)) {
+            throw new BadArgumentsException(MESSAGE_INVALID_ARGUMENTS);
+        }
+    }
+
+    private static void checkIsArrayIfHasBracketFormat(Options options)
+            throws BadArgumentsException {
+        if (options.getBracketType().isPresent() && options.getOutputFormat() != IoFormat.ARRAY) {
+            throw new BadArgumentsException(MESSAGE_INVALID_ARGUMENTS);
+        }
+    }
+
+    private static void checkExistingIntInputFromToOptionIsBigOrLittle(Options options)
+            throws BadArgumentsException {
+        if (options.getInputFormat() != IoFormat.INT) {
+            return;
+        }
+        Optional<IoOption> fromToOption = options.getInputFromToOption();
+
+        if (fromToOption.isPresent()) {
+            IoOption option = fromToOption.get();
+            if (option != IoOption.BIG && option != IoOption.LITTLE) {
+                throw new BadArgumentsException(MESSAGE_INVALID_ARGUMENTS);
+            }
+        }
+    }
+
+    private static void checkExistingIntOutputFromToOptionIsBigOrLittle(Options options)
+            throws BadArgumentsException {
+        if (options.getOutputFormat() != IoFormat.INT) {
+            return;
+        }
+        Optional<IoOption> fromToOption = options.getOutputFromToOption();
+        if (fromToOption.isPresent()) {
+            IoOption option = fromToOption.get();
+            if (option != IoOption.BIG && option != IoOption.LITTLE) {
+                throw new BadArgumentsException(MESSAGE_INVALID_ARGUMENTS);
+            }
+        }
+    }
```

Comment:
**suggestion:** Consider merging this method with the method above.

This method is almost same as the method above. You should consider using already resolved option as an argument instead of resolving it in this method. After changing this, you should be able to merge these 2 methods.